File: src/lib/scala/Slate.Common/src/main/scala/slate/common/logging/Logger.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.logging


object Logger {


  def parseLogLevel(level:String):LogLevel = {
    level match {
      case "debug" => Debug
      case "info"  => Info
      case "warn"  => Warn
      case "error" => Error
      case "fatal" => Fatal
      case _       => Debug
    }
  }
}

File: src/common/common/src/main/kotlin/kiit/common/log/Logger.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package kiit.common.log

import kiit.common.Ignore

abstract class Logger(
    open val level: LogLevel = LogLevel.Warn,
    open val name: String = "",
    open val logType: Class<*>? = null
) : LogSupport {

    fun isEnabled(level: LogLevel): Boolean = level >= this.level

    override val logger: Logger? by lazy { this }
    open val raw: Any? = null

    /**
     * Logs an entry
     *
     * @param level
     * @param msg
     * @param ex
     */
    @Ignore
    fun performLog(level: LogLevel, msg: String?, ex: Throwable?) {
        if(level >= this.level) {
            log(LogEntry(name, level, msg ?: "", ex))
        }
    }

    /**
     * Logs an entry
     *
     * @param level
     * @param ex
     */
    @Ignore
    fun performLog(level: LogLevel, msg:String?, callback: () -> String) {
        if(level >= this.level) {
            val label = msg ?: ""
            val output = callback()
            log(LogEntry(name, level, "$label : $output"))
        }
    }

    abstract fun log(entry: LogEntry)
}
